### DISCUSSION TITLE
How to add build commands for submodule

### DISCUSSION THREAD

**Discussion by dataexcess** [vercel/vercel#9128] [Help]
Hello! 

I have a git submodule in my project located at libs/trois
the .gitmodules file looks like this:

```
[submodule "libs/trois"]
	path = libs/trois
	url = https://github.com/dataexcess/trois.git
```

I need to execute 2 scripts of the gitmodule in order for it to work with my parent project code...
this is the package.json of the gitmodule:

```
  "scripts": {
    "dev": "vite",
    "types": "tsc",
    "build": "vue-tsc --noEmit && vite build",
    "rollup": "tsc && rollup -c"
  },
```

Specifically, I need to tell Vercel to run 'npm run build' and 'npm run rollup' in the gitmodule directory before running the main 'nuxt generate' back in the parent project directory.

How can I accomplish this? Using a custom build command in Vercel? There is no information on how to run submodule commands before main build command. Thank you!

**Comment by amyegan** [comment]
Hi @dataexcess! What are you currently using for the [Build & Development Settings](https://vercel.com/docs/concepts/deployments/configure-a-build#) on your project?

<img src="https://user-images.githubusercontent.com/8247105/210640962-ee0e371d-6c2a-48e1-ab0a-9b4776ac7425.png" alt="Screenshot of build and development settings using the Vite preset" />

**Comment by dataexcess** [comment]
Hello! I figured it out already :) All works, thank you!
